packages/runtime/test/transport/tokens.test.ts

```diff
@@ -94,3 +95,244 @@ describe("Personalized tokens", () => {
+        expect(loadResult).toBeSuccessful();
+        expect(loadResult.value.passwordProtection?.password).toBe("1234");
+        expect(loadResult.value.passwordProtection?.passwordIsPin).toBe(true);
+    });
+
+    test("error when loading a token with a wrong password", async () => {
+        const createResult = await runtimeServices1.transport.tokens.createOwnToken({
+            content: { key: "value" },
+            expiresAt: CoreDate.utc().add({ minutes: 10 }).toISOString(),
+            ephemeral: true,
+            passwordProtection: { password: "password" }
+        });
+        expect(createResult).toBeSuccessful();
+
+        const loadResult = await runtimeServices2.transport.tokens.loadPeerToken({ reference: createResult.value.truncatedReference, ephemeral: true, password: "wrong-password" });
+        expect(loadResult).toBeAnError(/.*/, "error.runtime.recordNotFound");
+    });
+
+    test("error when loading a token with no password", async () => {
+        const createResult = await runtimeServices1.transport.tokens.createOwnToken({
+            content: { key: "value" },
+            expiresAt: CoreDate.utc().add({ minutes: 10 }).toISOString(),
+            ephemeral: true,
+            passwordProtection: { password: "password" }
+        });
+        expect(createResult).toBeSuccessful();
+
+        const loadResult = await runtimeServices2.transport.tokens.loadPeerToken({
+            reference: createResult.value.truncatedReference,
+            ephemeral: true
+        });
+        expect(loadResult).toBeAnError(/.*/, "error.transport.noPasswordProvided");
+    });
+
+    test("validation error when creating a token with empty string as the password", async () => {
+        const createResult = await runtimeServices1.transport.tokens.createOwnToken({
+            content: { key: "value" },
+            expiresAt: CoreDate.utc().add({ minutes: 10 }).toISOString(),
+            ephemeral: true,
+            passwordProtection: { password: "" }
+        });
+        expect(createResult).toBeAnError("password must NOT have fewer than 1 characters", "error.runtime.validation.invalidPropertyValue");
+    });
+
+    test("validation error when creating a token with an invalid PIN", async () => {
+        const createResult = await runtimeServices1.transport.tokens.createOwnToken({
+            content: { key: "value" },
+            expiresAt: CoreDate.utc().add({ minutes: 10 }).toISOString(),
+            ephemeral: true,
+            passwordProtection: { password: "invalid-pin", passwordIsPin: true }
+        });
+        expect(createResult).toBeAnError(/.*/, "error.runtime.validation.invalidPin");
+    });
+});
+
+describe("Password-protected tokens for files", () => {
+    let fileId: string;
+
+    beforeAll(async () => {
+        fileId = (await uploadFile(runtimeServices1.transport)).id;
+    });
```

Comment:
please create a file for each test. Then you are way more flexible to test all combinations (see above for more context)

Reply:
I think you misunderstood this comment. I really meant just an **enmeshed file** not test.ts files, but I think it's ok like it is now?
@Milena-Czierlinski wdym?